### ISSUE TITLE
Question: Breaking changes between 1.92.0 and 2.0.0?

### ISSUE THREAD

**Issue by michaelholtermann** [pact-foundation/pact-ruby-standalone#101]
Sorry for bothering, but I wonder if there is any breaking change between 1.92.0 and 2.0.0, beside the file names of the release artifacts?

We are developing pact tests in Go, which are executed during CI and rely on `pact-ruby-standalone` (https://docs.pact.io/implementation_guides/go/readme#installation), so I assume the update won't break our tests or tool chain, but I'd like to be sure.

Thanks a lot!

**Comment by YOU54F** [comment]
file name changes, additional platforms and changes to the underlying build system for travelling ruby in order to move from Ruby 2.4 to 3.2.2

hopefully nothing should break but even with all the testing in the world, we won’t truly know until it’s in the hands of the users.

i assume you are using the non beta v1 version of pact go?

**Comment by YOU54F** [comment]
I'm not sure pact-go v2 beta uses the standalone, although it might do for broker related functionality, if it does help testing is always appreciated

I think with the jump from ruby 2.4 to 3.2.2 and the build system changes for macOS users, which previously used the x86_64 package under rosetta, and for windows users who now have both the 32 and 64 bit versions, it is warranted.

I'd like to honour semver rules, and thought I was doing a decent there with the major bump

**Comment by YOU54F** [comment]
It seems linux users (debian based at least) will need `libyaml-dev` from testing, otherwise they will see this error

```
libyaml-0.so.2: cannot open shared object file: No such file or directory -
 /tmp/cirrus-ci-build/standalone/linux-arm64-2.0.0/pact/lib/ruby/lib/ruby/3.2.0/aarch64-linux/psych.so
```

https://github.com/pact-foundation/pact-js-core/pull/445

It will be something that probably wants fixing in the packaging system (traveling ruby) but for now that seems to do the trick 👍🏾

**Comment by YOU54F** [comment]
## Problemo

Not all ruby gems packed in pact-ruby-standalone, are ruby 3.2 compatible, oddly this either hasn't shown up across several test suites and test, but appears to be reproducible in some circumstances

## Known errors on initial testing

- Pact-mock-service known to throw error on ruby 3.2 in some cases when writing the pact file 
    - affected parties
      -  pact-go v1 users https://github.com/pactflow/example-consumer-golang/actions/runs/4886667716/jobs/8722304524#step:4:44
      - homebrew pact standalone users
      - anyone using pact ruby standalone from latest
    - https://github.com/YOU54F/pact-mock_service/actions/runs/4888075739/jobs/8725395754#step:5:98
      - similar issue - recent discusses 2.7 -> 3.2 upgrade https://github.com/rapid7/metasploit-framework/issues/17593
      - Replicated in CI (linux) and own users machine (arm64 mac)    
    ```
    ERROR ThreadError: can't be called from trap context\n\t/Users/runner/hostedtoolcache/Ruby/3.2.2/x64/lib/ruby/3.2.0/timeout.rb:184:in `synchronize'
    ```
    

## Immediate remediation

1. ✅  Merge this change to install script - allowing users to pin version with install script
    1. https://github.com/pact-foundation/pact-ruby-standalone/pull/104
2. ✅ Set 2.0.0 release to pre-release 
    1. https://github.com/pact-foundation/pact-ruby-standalone/releases/tag/v2.0.0
3. 🚧 homebrew-pact-ruby-standalone requires a revert to pick up 1.92.0 or provide version specific formula
4. 🚧 applied changes to pact-python should probably be reverted

## Longer term resolution

1. Upgrade all relevant gems to 3.2

## List of packed ruby gems 

( I think this is the lot, pact-stub-service comes in the verifier)

- https://github.com/pact-foundation/pact_broker-client
- https://github.com/pact-foundation/pact-message-ruby
- https://github.com/pact-foundation/pact-mock_service
- https://github.com/pact-foundation/pact-provider-verifier


## Upgrade notes

- Pact-mock-service
    - needs at least fakefs gem updating as throws error on 3.0+

**Comment by gregtyler** [comment]
Hello! Is there any timeline on reverting the homebrew package? I've run into this error today and wondering if I need to revert to installing manually if homebrew's going to be out-of-action for a while.

**Comment by YOU54F** [comment]
done @gregtyler

**Comment by gregtyler** [comment]
Awezome, thanks so much Yousaf! 🌮

**Comment by YOU54F** [comment]
Hey @gregtyler 

Just to let you know I am have released a new version of [homebrew-pact-ruby-standalone](https://github.com/pact-foundation/homebrew-pact-ruby-standalone/pull/94)

if you have any issues, you’ll be able to stick with the v1.x branch with the following command

https://github.com/pact-foundation/homebrew-pact-ruby-standalone#previous-versions

**Comment by YOU54F** [comment]
2.0.3 seems pretty stable, we've had no more reports.

TL;DR - Breaking changes are the dropping of linux 32-bit and file name changes to the release artefacts (which might affect users scripts)

Thanks for raising @michaelholtermann - am going to close this one off now
